client/app.jsx

```diff
@@ -22,19 +29,35 @@ export default class App extends React.Component {
       .catch(err => console.error('GET error', err.message));
   }
 
-  render() {
+  renderPage() {
+    const prodId = Number(this.state.route.id);
```

Comment:
I wouldn't even bother making this variable until it was actually needed. It make the code more confusing since the reader has to remember that there is a variable floating around up here that might not even be relevant. Also, you can get it from the params as i mentioned in a previous comment.